src/main/java/co/cask/http/HttpMethodInfo.java

```diff
@@ -114,7 +115,8 @@ private void bodyConsumerChunk(ChannelBuffer buffer) throws HandlerException {
     try {
       bodyConsumer.chunk(buffer, responder);
     } catch (Throwable t) {
-      throw bodyConsumerError(t);
+      bodyConsumerError(t);
```

Comment:
It would be a good idea to wrap invocation of `consumer.handleError()` in a try blcok.